gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DiskStoreImpl.java

```diff
@@ -1909,7 +1915,11 @@ public void run() {
         // logger.info(LocalizedStrings.DEBUG, "DEBUG", ignore);
         // the above checkCancelInProgress will throw a CancelException
         // when we are being shutdown
-      } catch(Throwable t) {
+      } catch (Throwable t) {
+        if (!(t instanceof IOException)) {
```

Comment:
I think a cancelInProgress check can be added here too before throwing either DAE or GemFireIOE

Reply:
Yes. Will do.